core/protocols/aws-xml-protocol/src/main/java/software/amazon/awssdk/protocols/xml/internal/unmarshall/XmlProtocolUnmarshaller.java

```diff
@@ -79,19 +90,19 @@ public <TypeT extends SdkPojo> TypeT unmarshall(SdkPojo sdkPojo,
     }
 
     SdkPojo unmarshall(XmlUnmarshallerContext context, SdkPojo sdkPojo, XmlElement root) {
-        Optional<SdkField<?>> payloadMemberAsBlobType = getBlobTypePayloadMemberToUnmarshal(sdkPojo);
         for (SdkField<?> field : sdkPojo.sdkFields()) {
             XmlUnmarshaller<Object> unmarshaller = REGISTRY.getUnmarshaller(field.location(), field.marshallingType());
 
-            if (root != null && field.location() == MarshallLocation.PAYLOAD) {
+            if (field.location() == MarshallLocation.PAYLOAD) {
```

Comment:
Now that `root` can be null, it's hard for me to understand whether any uses of `root` below would cause a NPE. It spreads into a lot of different places. Is there any way to handle that case more explicitly? E.g. on line 108 just say `if (root == null) {` and handle it there?